### PR TITLE
Master node status in Fluence monitoring

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: node/src/main/scala/fluence/node/solvers/SolversPool.scala [hunk 1]
```diff
@@ -49,15 +49,17 @@ class SolversPool[F[_]: ContextShift: Timer](
   /**
    * Returns true if the solver is in the pool and healthy, and false otherwise. Also returns solver instance.
    */
-  private def checkSolverHealthy(params: SolverParams): F[(Boolean, Option[Solver[F]])] = {
+  private def checkSolverHealthy(params: SolverParams, isNewF: Ref[F, Boolean]): F[(Boolean, Option[Solver[F]])] = {
     for {
       map <- solvers.get
       oldSolver = map.get(params)
+      isNew <- isNewF.get
```
Comment (folex) [inline node/src/main/scala/fluence/node/solvers/SolversPool.scala]: That looks hacky. Maybe there is a way to refactor it to be more straightforward?

File: node/src/main/scala/fluence/node/solvers/SolversPool.scala [hunk 2]
```diff
@@ -49,15 +49,17 @@ class SolversPool[F[_]: ContextShift: Timer](
...
       healthy <- oldSolver match {
         case None => F.delay(false)
```
Comment (alari) [inline node/src/main/scala/fluence/node/solvers/SolversPool.scala]: It's pure, why to add `delay` overhead?

File: node/src/main/scala/fluence/node/solvers/SolversPool.scala [hunk 3]
```diff
@@ -49,15 +49,17 @@ class SolversPool[F[_]: ContextShift: Timer](
   /**
    * Returns true if the solver is in the pool and healthy, and false otherwise. Also returns solver instance.
    */
-  private def checkSolverHealthy(params: SolverParams): F[(Boolean, Option[Solver[F]])] = {
+  private def checkSolverHealthy(params: SolverParams, isNewF: Ref[F, Boolean]): F[(Boolean, Option[Solver[F]])] = {
```
Comment (alari) [inline node/src/main/scala/fluence/node/solvers/SolversPool.scala]: Could you add doc comment please? In what case `Solver` is returned, in what it doesn't? What does `isNewF` mean? How its value affects the function execution?

File: monitoring/src/fluence.ts [hunk 4]
```diff
@@ -15,31 +15,74 @@
...
-    let contract: Network = new web3js.eth.Contract(abi, contract_address) as Network;
+    let contractStatus = await getContractStatus(contract);
 
-    let status = await getStatus(contract);
-    document.body.innerHTML += `<div><pre>${JSON.stringify(status, undefined, 2)}</pre></div>`;
+    let responses = contractStatus.nodes.map((node) => {
+        // todo: `+400` is a temporary solution, fix it after implementing correct port management
+        let url = `http://${node.ip_addr}:${node.start_port + 400}/status`;
+        return axios.get(url).then((res) => {
+            res.data.status = "ok";
```
[the remark this answers is not in the document]
Comment (DieMyst) [reply monitoring/src/fluence.ts]: this is a mark that all ok or not with node. Without this, it is not clear if a node is available or not.

**Pull request merged**